src/plugins/discover/public/types.ts

```diff
@@ -6,14 +6,35 @@
  * Side Public License, v 1.
  */
 
-import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
-
-export type ElasticSearchHit<T = unknown> = estypes.SearchHit<T>;
-
-export type HitsFlattened = Array<Record<string, unknown>>;
+import * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
 
 export type ValueToStringConverter = (
   rowIndex: number,
   columnId: string,
   options?: { disableMultiline?: boolean }
 ) => { formattedString: string; withFormula: boolean };
+
+export type EsHitRecord = estypes.SearchHit;
+export type EsHitRecordList = EsHitRecord[];
+
+/**
+ * This is the record/row of data provided to our Data Table
+ */
+export interface DataTableRecord {
+  /**
+   * A unique id generated by index, id and routing of a record
+   */
+  id: string;
+  /**
+   * The document returned by Elasticsearch for search queries
+   */
+  raw: EsHitRecord;
+  /**
+   * A flattened version of the ES doc or data provided by SQL, aggregations ...
+   */
+  flattened: Record<string, unknown>;
+  /**
+   * Determines that the given doc is the anchor doc when rendering view surrounding docs
+   */
+  isAnchor?: boolean;
+}
```

Comment:
This is the new central type for our data displayed in both of our tables (legacy & document explorer).